pkg/api/types.go

```diff
@@ -134,12 +134,13 @@ type KubernetesConfig struct {
 
 // MasterProfile represents the definition of the master cluster
 type MasterProfile struct {
-	Count                    int    `json:"count"`
-	DNSPrefix                string `json:"dnsPrefix"`
-	VMSize                   string `json:"vmSize"`
-	VnetSubnetID             string `json:"vnetSubnetID,omitempty"`
-	FirstConsecutiveStaticIP string `json:"firstConsecutiveStaticIP,omitempty"`
-	Subnet                   string `json:"subnet"`
+	Count                    int               `json:"count"`
+	DNSPrefix                string            `json:"dnsPrefix"`
+	VMSize                   string            `json:"vmSize"`
+	VnetSubnetID             string            `json:"vnetSubnetID,omitempty"`
+	FirstConsecutiveStaticIP string            `json:"firstConsecutiveStaticIP,omitempty"`
+	Subnet                   string            `json:"subnet"`
+	Secrets                  []KeyVaultSecrets `json:"secrets,omitempty"`
```

Comment:
should I be adding these on the Linux/windows profiles instead? That way I could check more easily for if we should have a certificate store specified.  It would also mirror how we are handling ssh keys and admin passwords. We would lose a little control on where we are storing certs. thoughts?